### ISSUE TITLE
Huge memory usage with IM7

### ISSUE THREAD

**Issue by Watson1978** [rmagick/rmagick#817]
Ruby GC doesn't seem to work properly with ImageMagick 7.

With ImageMagick 6, Ruby GC works well by https://github.com/rmagick/rmagick/pull/697

Please confirm using script in https://github.com/rmagick/rmagick/tree/master/benchmarks/memory

![memory_usage](https://user-images.githubusercontent.com/199156/66251245-3d825c00-e789-11e9-92c1-59f2566e9239.png)

* Blue line : Running with ImageMagick-6.9.10-66
* Red line : Running with ImageMagick-7.0.8-66

(Actual results : https://gist.github.com/Watson1978/45dbc1c0a68027a4c16566d567daf960)

**Comment by Watson1978** [comment]
Seems ImageMagick 7 allocates the memory using `posix_memalign` by default.
```c
#if defined(MAGICKCORE_HAVE_POSIX_MEMALIGN)
  if (posix_memalign(&memory,alignment,extent) != 0)
    memory=NULL;
```
https://github.com/ImageMagick/ImageMagick/blob/850d5a9fc5d8d6ae7534421737a1709efe164a77/MagickCore/memory.c#L264-L266

It ignore the user custom memory handler set by `SetMagickMemoryMethods()`

If undefined `MAGICKCORE_HAVE_POSIX_MEMALIGN` in ImageMagick7, it will works well as  expectedly.

**Comment by dlemstra** [comment]
The same is happening inside ImageMagick 6 and I don't understand why the behaviour is different between IM7 and IM6. Both of them don't call the custom memory handler.

**Comment by Watson1978** [comment]
Seems ImageMagick's `configure` generates different configurations on macOS and Linux.
On Linux, it will use `posix_memalign()`...

**Comment by dlemstra** [comment]
Starting with ImageMagick `7.0.9-0` there will be a new method called `SetMagickAlignedMemoryMethods` that can be used to override the call to `posix_memalign`. But I suspect we will need to call xmalloc instead.

**Comment by Watson1978** [comment]
Ruby has `rb_aligned_malloc()` and `rb_aligned_free()` as private function. We might need to send Pull Request to make them public.

https://github.com/ruby/ruby/blob/ca7de0334d26b28eb8fd731b130404beb5cf5c12/gc.c#L9708-L9754

**Comment by Watson1978** [comment]
This should be solved by #836
